File: cpp/sorting/selection_sort.test.cpp
#include <iostream>

#include "selection_sort.hpp"

using namespace std;

int main() {
  int arr[] = {76, 34, 67, 32, 78, 45};
  const int length = sizeof(arr) / sizeof(*arr);
  SelectionSort<int> *selectionSort = new SelectionSort<int>();

  cout << "Before: " << selectionSort->toString(arr, length) << endl;
  selectionSort->sort(arr, length);
  cout << "After: " << selectionSort->toString(arr, length) << endl;

  int expected[] = {32, 34, 45, 67, 76, 78};
  for (int i = 0; i < length; i++) {
    assert(arr[i] == expected[i]);
  }

  cout << "Test passed" << endl;

  return 0;
}

File: python/sorting/selection_sort_test.py
from .sort import SortType
from .selection_sort import SelectionSort
from .perf_measurer import PerfMeasurer


def test_selection_sort():
    arr = [76, 34, 67, 32, 78, 45]
    sorter = SelectionSort()
    sorter.sort(arr)
    expected_asc = [32, 34, 45, 67, 76, 78]
    assert arr == expected_asc

    sorter.sort(arr, SortType.desc)
    expected_desc = [78, 76, 67, 45, 34, 32]
    assert arr == expected_desc

    perfMeasurer = PerfMeasurer(sorter)
    print(perfMeasurer.reportPerf())
